File: api/src/main/java/net/blueshell/api/config/RestClientConfig.java
package net.blueshell.api.config;

import org.springframework.boot.web.client.RestTemplateBuilder;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.client.RestTemplate;

@Configuration
public class RestClientConfig {

    /**
     * Central RestTemplate bean.
     * <p>
     * - Uses Spring’s {@link RestTemplateBuilder} so any
     *   auto-registered customisers (Jackson, timeouts, interceptors, etc.)
     *   are applied automatically.
     */
    @Bean
    public RestTemplate restTemplate(RestTemplateBuilder builder) {
        return builder.build();
    }
}

File: api/src/main/kotlin/net/blueshell/api/config/RestClientConfig.kt
package net.blueshell.api.config

import org.springframework.boot.web.client.RestTemplateBuilder
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.client.RestTemplate

@Configuration
class RestClientConfig {
    /**
     * Central RestTemplate bean.
     * 
     * 
     * - Uses Spring’s [RestTemplateBuilder] so any
     * auto-registered customisers (Jackson, timeouts, interceptors, etc.)
     * are applied automatically.
     */
    @Bean
    fun restTemplate(builder: RestTemplateBuilder): RestTemplate? {
        return builder.build()
    }
}

